### PR TITLE
Added X-Ray Connector.

### PR DESCRIPTION
**Pull request by bre1470**

### CODE REVIEW

File: src/XRay/XRayOptions.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    /**
+     * Data points for the x-ray.
+     */
+    dataPoints?: XRayDataPointOptions;
+
+    /**
+     * Array of portfolio holdings.
+     */
+    holdings?: (Array<MorningstarHoldingAmountOptions>|Array<MorningstarHoldingWeightOptions>);
```
Comment (Eskils) [inline src/XRay/XRayOptions.ts:163]: Very long, and written out in full multiple places. I would consider making a type alias for this.

File: test/unit-tests/XRay/Breakdown.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    });
+
+    Assert.ok(
+        connector instanceof MC.XRayConnector,
+        'Connector should be instance of TimeSeries class.'
+    );
+
+    Assert.ok(
+        connector.converter instanceof MC.XRayConverter,
+        'Converter should be instance of TimeSeries DividendSeriesConverter.'
```
Comment (Eskils) [inline test/unit-tests/XRay/Breakdown.test.ts:35]: Copy-paste error

**Review by Eskils — APPROVED** [review]
A thorough job on this connector. 🙌 My feedback is just nitpick.

File: src/XRay/XRayConnector.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            if (isMorningstarHoldingAmountOptions(holding)) {
+                amountHoldings.push(holding);
+            }
+            if (isMorningstarHoldingWeightOptions(holding)) {
+                weightHoldings.push(holding);
+            }
+        }
+
+        const bodyJSON: XRayPortfolioObject = {
+            benchmarkId: (options.benchmarkId || 'XIUSA0010V'),
```
[the remark this answers is not in the document]
Comment (bre1470) [reply src/XRay/XRayConnector.ts:190]: I have added a note to the `benchmarkId` option.

**Pull request merged**